lib/simplifyExpression/search.js

```diff
@@ -0,0 +1,127 @@
+import {print, parse} from 'math-parser';
+import {canApplyRule, applyRule, rules} from 'math-rules';
+import {Status} from '../Status.js';
+import simplifyExpressionString from './index.js';
+import SimplifyTypes from './SimplifyTypes.js';
+
+// Checks if rules from math-rules can be applied to nodes
+function apply(name, node){
+  const rule = rules[name];
+  if (canApplyRule(rule, node)){
+    const newNode = applyRule(rule, node);
+    return Status.nodeChanged(
+      name, node, newNode
+    );
+  }
+  return Status.noChange(node);
+}
+
+export function basics(node){
+  for (var rule in SimplifyTypes.BASIC_FUNCTIONS){
+    const nodeStatus = apply(rule, node);
+    if (nodeStatus.hasChanged()) {
+      return nodeStatus;
+    } else {
+      node = nodeStatus.newNode;
```

Comment:
I feel like I remember there being some reason we still need this - what is it? what fails when you don't update? shouldn't all changes trigger hasChanged?